presentation/src/main/java/com/example/user/test/presentation/app/MarvelApplication.kt

```diff
@@ -0,0 +1,14 @@
+package com.example.user.test.presentation.app
+
+import android.app.Application
+
+class MarvelApplication: Application() {
+
+    companion object {
+        lateinit var instance: MarvelApplication
```

Comment:
What is the purpose of this field? It is needed, you must make `set` private to prevent any initialization out of this class.